api/handlers.go

```diff
@@ -534,6 +565,18 @@ func (api *API) userPUT(w http.ResponseWriter, req *http.Request, _ httprouter.P
 	}
 
 	if payload.Password != "" {
+		// Check if the registrations are open. If they are not then changing
+		// passwords is also not allowed.
+		val, err := api.staticDB.ReadConfigValue(req.Context(), database.ConfValRegistrationsDisabled)
+		if err != nil && !errors.Contains(err, mongo.ErrNoDocuments) {
```

Comment:
Maybe it makes sense to return `nil` or default to `false`  in case the error equals `mongo.ErrNoDocuments`, and do this inside `ReadConfigValue`. Not sure if I like it myself yet but it could clean it up a bunch. I think we definitely want to DRY this up a bit it's a bunch of code to read through for such a simple thing.